pkg/logger/logger.go

```diff
@@ -103,51 +113,234 @@ func (l *Logger) log(logColor *color.Color, message string) {
 	}
 }
 
+// SetLogLevel sets the logger's minimum log level.
 func (l *Logger) SetLogLevel(logLevel LogLevel) error {
 	l.LogLevel = logLevel
 	return nil
 }
 
-func (l *Logger) Error(err error) {
-	if err != nil && l.LogLevel != LogLevelOff {
-		_, err2 := theme.Colors.Error.Fprintln(color.Error, err.Error()+"\n")
-		if err2 != nil {
-			color.Red("Error logging the error:")
-			color.Red("%s\n", err2)
-			color.Red("Original error:")
-			color.Red("%s\n", err)
-		}
-	}
-}
-
-// isLevelEnabled checks if a given log level should be enabled based on the logger's current level
+// isLevelEnabled checks if a given log level should be enabled based on the logger's current level.
 func (l *Logger) isLevelEnabled(level LogLevel) bool {
 	if l.LogLevel == LogLevelOff {
 		return false
 	}
 	return logLevelOrder[level] >= logLevelOrder[l.LogLevel]
 }
 
+// Error logs at ERROR level with different input types, supporting structured logging.
+// It accepts:
+// - String message: Error("event_name", "key", "value").
+// - Error object: Error(err, "key", "value").
+// - AtmosError: Error(atmosErr, "key", "value"). // atmosErr is of type *errors.AtmosError.
+func (l *Logger) Error(errOrMsg interface{}, keyvals ...interface{}) {
+	// Only proceed if the log level allows for error messages
+	if l.LogLevel == LogLevelOff {
+		return
+	}
+
+	// Use AtmosLogger to handle the structured error logging
+	atmosLogger := &AtmosLogger{Logger: log.New(os.Stderr)}
+	atmosLogger.Error(errOrMsg, keyvals...)
+}
+
+// Trace logs a message at TRACE level if the current log level permits.
 func (l *Logger) Trace(message string) {
 	if l.isLevelEnabled(LogLevelTrace) {
 		l.log(theme.Colors.Info, message)
 	}
 }
 
+// Debug logs a message at DEBUG level if the current log level permits.
 func (l *Logger) Debug(message string) {
 	if l.isLevelEnabled(LogLevelDebug) {
```

Comment:
Charmbracelet doesn't check if the level is enabled for us when we call `l.log`?